dspace-api/src/main/java/org/dspace/importer/external/ror/service/RorImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    @Override
+    public Collection<ImportRecord> getRecords(Query query) throws MetadataSourceException {
+        return retry(new SearchByQueryCallable(query));
+    }
+
+    @Override
+    public ImportRecord getRecord(Query query) throws MetadataSourceException {
+        List<ImportRecord> records = retry(new SearchByIdCallable(query));
+        return CollectionUtils.isEmpty(records) ? null : records.get(0);
+    }
+
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Query query) throws MetadataSourceException {
+        throw new MethodNotFoundException("This method is not implemented for ROR");
+    }
+
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Item item) throws MetadataSourceException {
+        throw new MethodNotFoundException("This method is not implemented for ROR");
+    }
+
+    @Override
+    public void init() throws Exception {
+    }
+
+    /**
+     * This class is a Callable implementation to get ADS entries based on query
+     * object. This Callable use as query value the string queryString passed to
+     * constructor. If the object will be construct through Query.class instance, a
+     * Query's map entry with key "query" will be used. Pagination is supported too,
+     * using the value of the Query's map with keys "start" and "count".
+     * 
+     * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+     */
+    private class SearchByQueryCallable implements Callable<List<ImportRecord>> {
+
+        private Query query;
+
+        private SearchByQueryCallable(String queryString) {
+            query = new Query();
+            query.addParameter("query", queryString);
+        }
+
+        private SearchByQueryCallable(Query query) {
+            this.query = query;
+        }
+
+        @Override
+        public List<ImportRecord> call() throws Exception {
+            return search(query.getParameterAsClass("query", String.class));
+        }
+    }
+
+    /**
+     * This class is a Callable implementation to get an ADS entry using bibcode The
+     * bibcode to use can be passed through the constructor as a String or as
+     * Query's map entry, with the key "id".
+     *
+     * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+     */
+    private class SearchByIdCallable implements Callable<List<ImportRecord>> {
+        private Query query;
+
+        private SearchByIdCallable(Query query) {
+            this.query = query;
+        }
+
+        private SearchByIdCallable(String id) {
+            this.query = new Query();
+            query.addParameter("id", id);
+        }
+
+        @Override
+        public List<ImportRecord> call() throws Exception {
+            return searchById(query.getParameterAsClass("id", String.class));
+        }
+    }
+
+    /**
+     * This class is a Callable implementation to count the number of entries for an
+     * ADS query. This Callable use as query value to ADS the string queryString
```

Comment:
These Javadocs also reference ADS.  It looks like you may have reused some of the ADS code in general (which is fine). But, we should then remove any references to ADS please